Develop

SaveliiSidorkin opened the pull request:
Constants - константы
Courier - сериализация курьера
CourierClient  - запросы создания/авторизации/удаления курьера
CourierCredentials - авторизация курьера
CourierGenerator - генерация данных тестового курьера
Order - сериализация заказа
OrderClient - запросы создания/получения заказа

Тесты
CourierClientTest - тесты на создание курьера
CourierLoginTest - тесты на авторизацию курьера
OrderCreateTest - тесты на создание заказа
OrderListTest - тесты на получение списка заказов

Commit:
add Courier's and Order's files

Commit:
add allure reports

Commit:
change name of step's

alesyaisachenko commented on src/main/java/courier/Courier.java:
```diff
@@ -0,0 +1,40 @@
+package courier;
+
+public class Courier {
```
Можно лучше:
можно использовать [@DaTa](https://github.com/DaTa) из библиотеки Lombok

alesyaisachenko commented on src/main/java/courier/CourierClient.java:
```diff
@@ -0,0 +1,39 @@
+
+public class CourierClient extends Constants {
+    @Step("Отправка запроса создания курьера POST /api/v1/courier")
+    public ValidatableResponse create(Courier courier) {
```
Можно лучше:

create(Courier courier) - лучше  createCourier(Courier courier), это упрощает чтение кода теста, в этом случае сразу из кода ясно какой именно объект создается.

alesyaisachenko commented on src/main/java/courier/CourierClient.java:
```diff
@@ -0,0 +1,39 @@
+
+import static io.restassured.RestAssured.given;
+
+public class CourierClient extends Constants {
```
Это нужно доработать:

наследование от класса Constants нужно убрать.

в классе Constant хранятся URL.

в классе CourierClient хранятся методы для формирования и отправки запроса. 

У этих классов разное назначение, поэтому наследование использовать нельзя.

alesyaisachenko commented on src/main/java/order/OrderClient.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @Step("Отмена заказа PUT /api/v1/orders/cancel")
+    public ValidatableResponse canselOrder(int track) {
```
Это нужно доработать:

в теле метода не используется параметр int track

alesyaisachenko commented on src/test/java/courierTest/CourierClientTest.java:
```diff
@@ -0,0 +1,96 @@
+package courierTest;
```
Это нужно доработать:
 courierTest -- нужно  couriertest , пакеты именуются в таком формате.

Подправь, плиз, этот момент во всем коде

alesyaisachenko commented on src/test/java/courierTest/CourierClientTest.java:
```diff
@@ -0,0 +1,96 @@
+    @Test
+    @DisplayName("Создание курьера")
+    @Description("Создаем курьера и логинимся")
+    public void CourierCanBeCreateTest() {
```
Это нужно доработать:

 CourierCanBeCreateTest()  -- нужно  courierCanBeCreateTest() методы именуются в таком формате.

Подправь, плиз, этот момент во всем коде.

alesyaisachenko commented on src/test/java/courierTest/CourierClientTest.java:
```diff
@@ -0,0 +1,96 @@
+        boolean status = response.extract().path("ok");
+        assertTrue(status);
+        ValidatableResponse loginResponce = courierClient.login(new CourierCredentials(courier.getLogin(), courier.getPassword()));
+        loginResponce.statusCode(200);
```
Можно лучше:

статус коды можно было взять из библиотеки:
import  static  org.apache.http.HttpStatus.*;

statusCode(SC_CREATED)

Commit:
-Переименовал методы
-Вынес проверку отмены заказа в отдельный тест
-Переименовал пакеты
-Вынес метод getSpec в отдельный класс
-Добавил библеотеку httpStatus

SaveliiSidorkin commented on src/main/java/order/OrderClient.java:
```diff
@@ -22,16 +24,16 @@ public ValidatableResponse getListOrder() {
 
     @Step("Отмена заказа PUT /api/v1/orders/cancel")
-    public ValidatableResponse canselOrder(int track) {
+    public ValidatableResponse cancelOrder(int track) {
```
Оставил по рекомендации наставника и добавил отдельный тест на отмену заказа:
"В этом тесте мы просто используем этот эндпоинт в соответствии с документацией. Если эндпоинт не работает в соответствии с документацией, это должен показать отдельный тест на этот эндпоинт"
По ТЗ в теле метода передается параметр track
